woocommerce/woocommerce
# Fix Customize Your Store UI bugs

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/team-ghidorah/issues/265

This PR fixes the following UI bugs:

1. (Safrai) Uneven focus, looks like it's a bit more on one side than the other

<img src="https://github.com/woocommerce/team-ghidorah/assets/27843274/fbf692e6-60ee-4bd0-a49e-a14fd3436f5d" width="400px" />
<img src="https://github.com/woocommerce/team-ghidorah/assets/27843274/3217dbbc-c7b3-4c03-88d9-f8920d3e630f" width="450px" />

2. Frame spinner not centered

<img src="https://github.com/woocommerce/team-ghidorah/assets/27843274/e4b6c4b4-ad07-4ab1-b174-cd7096633f52" width="550px" />

3. The 'x' on the spotlight doesn't seem to be clickable


<img src="https://github.com/woocommerce/team-ghidorah/assets/27843274/ef177909-9ee8-47b3-ba73-1e7398669b75" width="550px" />


4. AI API call loader looks a bit janky

https://github.com/woocommerce/team-ghidorah/assets/27843274/48b37da7-cbe0-49af-8ecc-53edea21e25c

5. Sidebar section group title spacing

<img src="https://github.com/woocommerce/woocommerce/assets/4344253/654ea178-ae9a-490e-bd48-c0b2777ef2ed" width="550px" />

7. Cannot save cys logo changes

<img src="https://github.com/woocommerce/woocommerce/assets/4344253/ea75fcec-94c8-42fd-a0a1-db320422038e" width="550px" />

8. Missing selected pattern border

9. Added `woocommerce_customize_store_ai_suggestions` to the option API allowed list.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:


This feature is part of the larger customize your store development and behind a feature flag, hence QA review can be deferred until the call for testing before release.

Make sure you are testing this with a **Jetpack connected** instance of WooCommerce, as the AI text completion endpoint can only be called with a Jetpack connected account.

1. Create a new WooCommerce installation with this version.
2. Make sure to enable `customize-store` feature flag
3. Use Safrai
4. Go to `WooCommerce -> Home` and click `Customize Store` task
5. Click `Design with AI` button
6. Click on the text area
7. Observe that the focus style is even

![Screenshot 2023-09-28 at 14 09 49](https://github.com/woocommerce/woocommerce/assets/4344253/99dc8af8-c139-47d3-a2f5-86617160a1db)

8. Follow through the flow all the way till the loader shows up
9. Observer that the AI API call loader is smoothly
10. Go to Assembler Hub
11. Observe that the preview frame spinner is centered
12. Click on "Take a tour" on the welcome modal
13. Confirm the `x` button is clickable
14. Observe that there is 32px spacing between "LAYOUT" and "Change fonts"
15. Go to "Change the color palette" (/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fassembler-hub%2Fcolor-palette)
16. Hover on the color palette option
17. Observe that the focus style is even
18. Go to "Change your header" (`/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fassembler-hub%2Fheader`)
19. Observe that the AI-selected header pattern on the sidebar has an active border color style.
20. Go to "Change your header" (`/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fassembler-hub%2Fheader`)
21. Observe that the AI-selected header pattern on the sidebar has an active border color style.
22. Go to `Change your homepage` (`/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fassembler-hub%2Fhomepage`)
23. Observe that the AI-selected header pattern on the sidebar has an active border color style.
24. Add a logo
25. Click on "Save"
26. Go to `Add your logo` again
27. Change logo settings
28. Confirm the "Save" button is clickable.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
